### DISCUSSION TITLE
Error Occurred: Handlebars Issue When Sending Organization Invitations

### DISCUSSION THREAD

**Discussion by stitooo2023** [dani-garcia/vaultwarden#5085] [Q&A] [answered]
<p>Hi, I'm experiencing an issue when inviting a user to an organization in Vaultwarden. Here are the details:</p>

<h2>Issue Description</h2>
<p>When trying to invite a user to an organization, I encounter the following error message in the web interface:</p>

<pre>
An error has occurred.
Handlebars
</pre>

<p>In the logs, I see:</p>

<pre>
[2024-10-15 13:22:59.646][request][INFO] POST /api/organizations/xxxxx-15e5-4a81-xxxx-c130402b1a13/users/invite
[2024-10-15 13:22:59.648][error][ERROR] Handlebars.
[CAUSE] RenderError {
    template_name: Some("email/send_org_invite.html"),
    line_no: Some(12),
    column_no: Some(65),
    reason: MissingVariable(Some("org_id")),
    unimplemented: false,
}
[2024-10-15 13:22:59.648][response][INFO] (send_invite) POST /api/organizations/&lt;org_id&gt;/users/invite => 400 Bad Request
</pre>

<p>It seems like the <code>org_id</code> is missing when the <code>send_org_invite.html</code> template is being rendered, causing a <code>RenderError</code>.</p>

<h2>Environment Details</h2>
<ul>
    <li>Vaultwarden version: v1.32.2</li>
    <li>Web-vault version: v2024.6.2c</li>
    <li>OS/Arch: linux/x86_64</li>
    <li>Running within a container: true (Base: Debian)</li>
    <li>Environment settings overridden: true</li>
    <li>Uses a reverse proxy: false</li>
    <li>Internet access: true</li>
    <li>Internet access via a proxy: false</li>
    <li>DNS Check: true</li>
    <li>Browser/Server Time Check: true</li>
    <li>Server/NTP Time Check: true</li>
    <li>Domain Configuration Check: false</li>
    <li>HTTPS Check: true</li>
    <li>Database type: SQLite</li>
    <li>Database version: 3.46.0</li>
</ul>

<h2>Config (Generated via diagnostics page)</h2>
<details>
<summary>Show Running Config</summary>

<p><strong>Environment settings which are overridden:</strong> DOMAIN, SENDS_ALLOWED, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, SIGNUPS_VERIFY_RESEND_TIME, SIGNUPS_VERIFY_RESEND_LIMIT, EMERGENCY_ACCESS_ALLOWED, EMAIL_CHANGE_ALLOWED, PASSWORD_HINTS_ALLOWED, SHOW_PASSWORD_HINT, ADMIN_TOKEN</p>

<pre>
{
  "domain": "*****://******************************",
  "extended_logging": true,
  "push_enabled": true,
  "signups_verify": true,
  "smtp_host": "*************************",
  "smtp_from": "******************************",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "admin_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "smtp_username": "******************************",
  "smtp_password": "***",
  "emergency_access_allowed": true,
  "web_vault_enabled": true,
  "signups_verify_resend_limit": 6,
  "admin_token": "***",
  "org_creation_users": "***********************",
  "org_events_enabled": false,
  "data_folder": "data",
  "enable_websocket": true,
  "email_2fa_auto_fallback": false,
  "icon_blacklist_non_global_ips": true,
  "password_iterations": 600000,
  "show_password_hint": false
}
</pre>
</details>

<h2>Docker Configuration</h2>

<p>Here is my <code>docker-compose.yaml</code>:</p>

<pre>
services:
  vaultwarden:
    image: vaultwarden/server:latest
    container_name: vaultwarden
    environment:
      - ROCKET_TLS={certs="/ssl/bitwarden/bitwarden.crt",key="/ssl/bitwarden/bitwarden.key"}
      - DOMAIN=https://test.eu
      - LOGIN_RATELIMIT_MAX_BURST=10
      - LOGIN_RATELIMIT_SECONDS=60
      - ADMIN_RATELIMIT_MAX_BURST=10
      - ADMIN_RATELIMIT_SECONDS=60
      - ADMIN_TOKEN={VAULTWARDEN_ADMIN_TOKEN}
      - SENDS_ALLOWED=false
      - EMERGENCY_ACCESS_ALLOWED=true
      - WEB_VAULT_ENABLED=true
      - SIGNUPS_ALLOWED=true
      - SIGNUPS_VERIFY=true
      - SIGNUPS_VERIFY_RESEND_TIME=3600
      - SIGNUPS_VERIFY_RESEND_LIMIT=5
      - PUSH_ENABLED=true
      - PUSH_INSTALLATION_ID=****
      - PUSH_INSTALLATION_KEY=****
      - PUSH_RELAY_URI=https://api.bitwarden.eu
      - PUSH_IDENTITY_URI=https://identity.bitwarden.eu
      - EXTENDED_LOGGING=true
      - LOG_FILE=/data/log/vaultwarden.log
      - PASSWORD_HINTS_ALLOWED=false
      - SHOW_PASSWORD_HINT=false
      - EMAIL_CHANGE_ALLOWED=false
      - ORG_CREATION_USERS=test@test.eu
      - TEMPLATES_FOLDER=/data/templates
      - TZ=Europe/Warsaw
    volumes:
      - /docker/vaultwarden/data:/data/
      - /docker/ssl/:/ssl/
    restart: unless-stopped
    ports:
      - "7070:80"
</pre>

<h2>Additional Information</h2>
<ul>
<li>I have set up a custom <code>TEMPLATES_FOLDER</code> at <code>/data/templates</code>, and the <code>send_org_invite.html</code> template is present there.</li>
<li>I double-checked that the <code>org_id</code> is correctly set in the organization data and in the invitation request.</li>
<li>The email configuration and TLS certificates seem to be working correctly, as other email functions do not produce errors.</li>
<li>I've obtained the template from the following page:
<p><a href="https://github.com/dani-garcia/vaultwarden/wiki/Translating-the-email-templates">Translating the email templates</a></p>
</li>
<li>I also tried running the <code>docker-compose.yaml</code> with the commented line:
<pre><code>#- TEMPLATES_FOLDER=/data/templates</code></pre>
<p>However, the problem still persists.</p>
</li>
</ul>

**Comment by BlackDex** [comment, marked as answer]
Duplicate of #5068 and #4917

Update your templates to match with the latest version of the version you are running https://github.com/dani-garcia/vaultwarden/tree/1.32.2/src/static/templates/email

Since you modified the configuration via the Vaultwarden Admin Backend `/admin`, Vaultwarden is using the stored config in `config.json` and not (all) your env variables as you can see in the Support String if you open `Show Running Config`:

> Environment settings which are overridden: 

Update the templates and it will work again.

**Reply by stitooo2023** [reply]
Thank you for your guidance!

Updating the templates send_org_invite.hbs and send_org_invite.html.hbs resolved the issue, and everything is working perfectly now.

I truly appreciate your help!
